app/models/manageiq/providers/redhat/infra_manager/refresh_parser.rb

```diff
@@ -334,9 +340,28 @@ def self.host_inv_to_network_hashes(inv, guest_device_uids)
   def self.host_inv_to_os_hash(inv, hostname)
     return nil if inv.nil?
 
-    result = {:name => hostname}
-    result[:product_name] = 'linux'
-    result
+    {
+      :name         => hostname,
+      :product_type => 'linux',
+      :product_name => extract_host_os_name(inv),
+      :version      => extract_host_os_full_version(inv[:os])
+    }
+  end
+
+  def self.extract_host_os_full_version(host_os)
```

Comment:
@masayag It's still missing a check whether `inv[:os]` is `nil` before calling.  For instance, in `extract_host_os_name`, it verifies that `inv[:os]` is not nil before calling `extract_host_os_full_version`.  And, `extract_host_os_full_version` assumes the input is not `nil`.

This might be OK.  I.e., perhaps it's impossible for `inv[:os]` to be `nil`.  I just noticed that one path guards for that specific case and one path doesn't.